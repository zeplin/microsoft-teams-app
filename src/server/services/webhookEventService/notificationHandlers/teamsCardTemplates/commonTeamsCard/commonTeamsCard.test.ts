import { commonTeamsCard } from "./index";

describe("commonTeamsCard", () => {
    it("should match snapshot when rendered with only required elements", () => {
        expect(
            commonTeamsCard({
                text: "text"
            })
        ).toMatchSnapshot();
    });

    it("should match snapshot when rendered with section", () => {
        expect(
            commonTeamsCard({
                text: "text",
                section: {
                    title: "sectionTitle",
                    text: "sectionText"
                }
            })
        ).toMatchSnapshot();
    });

    it("should match snapshot when rendered with images", () => {
        expect(
            commonTeamsCard({
                text: "text",
                images: ["http://placehold.it/400"]
            })
        ).toMatchSnapshot();
    });

    it("should match snapshot when rendered with links", () => {
        expect(
            commonTeamsCard({
                text: "text",
                links: [{
                    title: "Open in me",
                    url: "https://ergun.sh"
                }]
            })
        ).toMatchSnapshot();
    });

    it("should match snapshot when rendered with title", () => {
        expect(
            commonTeamsCard({
                title: "title",
                text: "text"
            })
        ).toMatchSnapshot();
    });

    it("should match snapshot when rendered with all elements", () => {
        expect(
            commonTeamsCard({
                title: "Project name",
                text: "**sertac ðŸŒ®** added a new comment on _Manage Zeplin Connector_ screen. ðŸ—ž",
                section: {
                    title: "Section title",
                    text: "Naptin nettin nettin naptin?"
                },
                links: [{
                    title: "Open in Web",
                    url: "https://ergun.sh"
                }, {
                    title: "Open in App",
                    url: "https://ergun.sh"
                }],
                images: ["http://placehold.it/160"]
            })
        ).toMatchSnapshot();
    });
});
