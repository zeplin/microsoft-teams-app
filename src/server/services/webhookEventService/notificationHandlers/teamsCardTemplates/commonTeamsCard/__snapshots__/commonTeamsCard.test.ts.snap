// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`commonTeamsCard should match snapshot when rendered with all elements 1`] = `
{
  "@type": "MessageCard",
  "@context": "https://schema.org/extensions",
  "title": "Project name",
  "text": "**sertac ðŸŒ®** added a new comment on _Manage Zeplin Connector_ screen. ðŸ—ž",
  "sections": [
    {
      "title": "Section title",
      "text": "Naptin nettin nettin naptin?",
      "images": [
        {
          "image": "http://placehold.it/160",
          "title": "Section Image"
        }
      ]
    }
  ],
  "potentialAction": [
    {
      "@type": "OpenUri",
      "name": "Open in Web",
      "targets": [
        {
          "os": "default",
          "uri": "https://ergun.sh"
        }
      ]
    },
    {
      "@type": "OpenUri",
      "name": "Open in App",
      "targets": [
        {
          "os": "default",
          "uri": "https://ergun.sh"
        }
      ]
    }
  ]
}
`;

exports[`commonTeamsCard should match snapshot when rendered with images 1`] = `
{
  "@type": "MessageCard",
  "@context": "https://schema.org/extensions",
  "text": "text",
  "sections": [
    {
      "images": [
        {
          "image": "http://placehold.it/400",
          "title": "Section Image"
        }
      ]
    }
  ]
}
`;

exports[`commonTeamsCard should match snapshot when rendered with links 1`] = `
{
  "@type": "MessageCard",
  "@context": "https://schema.org/extensions",
  "text": "text",
  "sections": [],
  "potentialAction": [
    {
      "@type": "OpenUri",
      "name": "Open in me",
      "targets": [
        {
          "os": "default",
          "uri": "https://ergun.sh"
        }
      ]
    }
  ]
}
`;

exports[`commonTeamsCard should match snapshot when rendered with only required elements 1`] = `
{
  "@type": "MessageCard",
  "@context": "https://schema.org/extensions",
  "text": "text",
  "sections": []
}
`;

exports[`commonTeamsCard should match snapshot when rendered with section 1`] = `
{
  "@type": "MessageCard",
  "@context": "https://schema.org/extensions",
  "text": "text",
  "sections": [
    {
      "title": "sectionTitle",
      "text": "sectionText",
      "images": []
    }
  ]
}
`;

exports[`commonTeamsCard should match snapshot when rendered with title 1`] = `
{
  "@type": "MessageCard",
  "@context": "https://schema.org/extensions",
  "title": "title",
  "text": "text",
  "sections": []
}
`;
